pubsub.ts

```diff
@@ -21,68 +19,101 @@ export type RedisPubSubMessage = {
 };
 
 class RedisSubscriptionImpl implements RedisSubscription {
-  private _isClosed = false;
+
+  get isConnected(): boolean {
+    return this.connection.isConnected;
+  }
+
   get isClosed(): boolean {
-    return this._isClosed;
+    return this.connection.isClosed;
   }
 
   private channels = Object.create(null);
   private patterns = Object.create(null);
 
-  constructor(private writer: BufWriter, private reader: BufReader) {}
+  constructor(private connection: RedisConnection) {
+    // Force retriable connection for connection shared for pub/sub.
+    if (connection.maxRetryCount === 0) connection.maxRetryCount = 10;
+  }
 
   async psubscribe(...patterns: string[]) {
-    await sendCommand(this.writer, this.reader, "PSUBSCRIBE", ...patterns);
+    await sendCommand(
+      this.connection.writer! as BufWriter,
```

Comment:
I think this type assertion is unnecessary.
```suggestion
      this.connection.writer!,
```